tests/node_pool_deployer.py

```diff
@@ -0,0 +1,120 @@
+import re
+
+import tests.utils as ut
+
+
+class NodePoolDep:
+    resources = ["bootstrap", "client"]
+    gcloud_delete = "yes | gcloud container node-pools delete {pool_name} --cluster={cluster_name} --zone={zone}"
+    gcloud_cmd = 'yes | gcloud beta container --project "{project_name}" node-pools create "{pool_name}" ' \
+                 '--cluster "{cluster_name}" --zone "{zone}" --node-version "{node_version}" ' \
+                 '--machine-type "custom-{cpu}-{mem}" --image-type "COS" --disk-type "{disk_type}" ' \
+                 '--disk-size "{disk_size}" --node-labels {labels} --metadata disable-legacy-endpoints=true ' \
+                 '--node-taints {taints} --scopes "https://www.googleapis.com/auth/devstorage.read_only",' \
+                 '"https://www.googleapis.com/auth/logging.write",' \
+                 '"https://www.googleapis.com/auth/monitoring",' \
+                 '"https://www.googleapis.com/auth/servicecontrol",' \
+                 '"https://www.googleapis.com/auth/service.management.readonly",' \
+                 '"https://www.googleapis.com/auth/trace.append" ' \
+                 '--num-nodes "{num_nodes}" --enable-autoscaling --min-nodes "{min_nodes}" --max-nodes "{max_nodes}" ' \
+                 '--no-enable-autoupgrade --enable-autorepair --max-surge-upgrade 1 --max-unavailable-upgrade 0'
+
+    def __init__(self, testconfig):
+        self.testconfig = testconfig
+        self.namespace = testconfig["namespace"]
+        self.pool_name = f"pool-{self.namespace}"
+        self.cluster_name = "spacemesh-cluster-elk"
+        self.zone = "us-west1-a"
+        self.default_config = {
+            "project_name": "spacemesh-198810",
```

Comment:
This (and the `zone`, above) should probably also not be hardcoded, this should probably be passed in from the env so it's easier to change, or stored in a config file or something